ceaos/sensors/temperature_sensor.py

```diff
@@ -0,0 +1,39 @@
+"""
+This file contains a class for a temperature sensor
+"""
+from .sensor_definition import Sensor  
+from ceaos.loggers.InfluxDB import InfluxDBConnection, InfluxDBLogger
+
+
+class Temperature(Sensor):
+
+    datatype = "General_Temp"  # static variable for Temperature class
+
+    def __init__(self, temp_value=None, datatype=datatype) -> None:
+        self.set_value(temp_value)
+        self.set_datatype(datatype) # defaults to "General_Temp" (i.e. instead of Air_Temp)
+
+    def set_value(self, temp_value: float):
```

Comment:
See the ceaos/sensors/sensor_definition.py file for the minimum implementation of the sensor. Seems like the code in the test would be appropriate for the read_sensor method here. You would need to make sure that the pins used are initialized in the object during its creation.

Reply:
Good call, will move that code into the read_value (I think that's what you meant by read_sensor?) method

Reply:
Hey, noticing that the set_value method is still live. Make sure that this interface implements the Sensor interface

Reply:
Could you clarify this a bit more? Do I use the code for "set_value" in the sensor_definition.py file, instead of temperature_sensor.py? Or do I do something else in the temperature_sensor.py file?

Sorry, I think I'm missing what you mean by "implements the Sensor interface". For context, I followed the approach in artificial_sensor.py, minus the calibrate method.

Reply:
So, the reason that artificial sensors have a set value method is because they are artificial, and we can set the value of the artificial sensor. The sensor interface only has a `read_value` and `calibrate` method. If you think about the sensor as an abstraction that measures some environmental variables, it doesn't make sense to be able to set the value, right? This object should not have a set_value function, only a `read_value` function. When something uses this sensor, it should be able to get up to date information by only calling the `read_value` function.

To add more to this, this pattern is OK to use for the Artificial Camera, because as an artificial sensor, it must also have its value set to return something.

Reply:
Ah I see, this makes a lot of sense now! I wasn't fully grasping the abstraction of an artificial vs real sensor, but this clears up my confusion. Many thanks, I'll update the code.